Source/WebCore/editing/Editing.cpp

```diff
@@ -877,7 +878,16 @@ int caretMinOffset(const Node& node)
 {
     auto* renderer = node.renderer();
     ASSERT(!node.isCharacterDataNode() || !renderer || renderer->isRenderText());
-    return renderer ? renderer->caretMinOffset() : 0;
+
+    if (renderer && renderer->isRenderText())
+        return renderer->caretMinOffset();
+
+    if (RefPtr pictureElement = dynamicDowncast<HTMLPictureElement>(node)) {
+        if (RefPtr firstImage = childrenOfType<HTMLImageElement>(*pictureElement).first())
+            return firstImage->computeNodeIndex();
```

Comment:
I think early returns instead of nested conditions would look a bit better, but that's just imo